tlslite/tlsrecordlayer.py

```diff
@@ -722,6 +722,19 @@ def _getMsg(self, expectedType, secondaryType=None, constructorType=None):
                                 yield result
                             continue
 
+                    # If we received a hearbeat request ...
+                    if recordHeader.type == ContentType.heartbeat:
```

Comment:
I mean, if you receive that message type, and you have not negotiated heartbeat through extensions, the response to that should be `unexpected_message`, so we should set some flag in TLSRecordLayer to remind us